contracts/limit_order_uniswap_v2.vy

```diff
@@ -97,102 +78,84 @@ def _safe_transfer_from(_token: address, _from: address, _to: address, _value: u
-    assert amount1_max > amount1_min
+    if token1 == VETH:
+        _path[last_index] = WETH
+    amounts: DynArray[uint256, MAX_SIZE] = UniswapV2Router(ROUTER).swapExactTokensForTokens(amount0, min_amount1, _path, self, block.timestamp)
+
     deposit_id: uint256 = self.deposit_size
-    pool: address = UniswapV2Factory(FACTORY).getPair(token0, token1)
     self.deposits[deposit_id] = Deposit({
-        token0: token0,
-        token1: token1,
-        amount0: amount0,
-        amount1_min: amount1_min,
-        amount1_max: amount1_max,
-        pool: pool,
+        path: path,
+        amount1: amounts[last_index],
         depositor: msg.sender
     })
     self.deposit_size = deposit_id + 1
-    log Deposited(deposit_id, token0, token1, amount0, amount1_min, amount1_max, pool, msg.sender)
-
-@external
-def cancel(deposit_id: uint256):
-    deposit: Deposit = self.deposits[deposit_id]
-    assert msg.sender == deposit.depositor
-    self.deposits[deposit_id] = Deposit({
-        token0: empty(address),
-        token1: empty(address),
-        amount0: 0,
-        amount1_min: 0,
-        amount1_max: 0,
-        pool: empty(address),
-        depositor: empty(address)
-    })
-    if deposit.token0 == VETH:
-        WrappedEth(WETH).withdraw(deposit.amount0)
-        send(msg.sender, deposit.amount0)
-    else:
-        self._safe_transfer(deposit.token0, msg.sender, deposit.amount0)
-    log Withdrawn(deposit_id, msg.sender, False, 0)
+    log Deposited(deposit_id, token0, token1, amount0, amounts[last_index], msg.sender, profit_taking, stop_loss)
 
 @internal
-def _withdraw(deposit_id: uint256, profit_taking_or_stop_loss: bool):
+def _withdraw(deposit_id: uint256, min_amount0: uint256, withdraw_type: WithdrawType):
     deposit: Deposit = self.deposits[deposit_id]
+    if withdraw_type == WithdrawType.CANCEL:
+        assert msg.sender == deposit.depositor
     self.deposits[deposit_id] = Deposit({
-        token0: empty(address),
-        token1: empty(address),
-        amount0: 0,
-        amount1_min: 0,
-        amount1_max: 0,
-        pool: empty(address),
+        path: empty(DynArray[address, MAX_SIZE]),
+        amount1: empty(uint256),
         depositor: empty(address)
     })
-    assert deposit.amount0 > 0
-    if deposit.token0 == VETH:
-        self._safe_approve(WETH, ROUTER, deposit.amount0)
-    else:
-        self._safe_approve(deposit.token0, ROUTER, deposit.amount0)
-    amounts: DynArray[uint256, 2] = [0, 0]
-    if deposit.token1 == VETH:
-        if profit_taking_or_stop_loss:
-            amounts = UniswapV2Router(ROUTER).swapExactTokensForETH(deposit.amount0, deposit.amount1_max, [deposit.token0, WETH], deposit.depositor, block.timestamp)
-        else:
-            amounts = UniswapV2Router(ROUTER).swapExactTokensForETH(deposit.amount0, deposit.amount1_min, [deposit.token0, WETH], deposit.depositor, block.timestamp)
+    assert deposit.amount1 > 0
```

Comment:
This assert is above in the diff as `assert deposit.amount0 > 0`. I'm not actually sure what `amount0` and `amount 1` represent semantically though.

Reply:
Because the swap path is reverted.